providers/implementations/encode_decode/encode_key2any.c

```diff
@@ -0,0 +1,908 @@
+                                        OSSL_CORE_BIO *cout,            \
+                                        OSSL_PASSPHRASE_CALLBACK *cb,   \
+                                        void *cbarg)                    \
+    {                                                                   \
+        struct key2any_ctx_st *ctx = vctx;                              \
+        int selection = type##_##kind##_selection;                      \
+        void *key = ossl_prov_import_key(impl##_keymgmt_functions,      \
+                                         ctx->provctx, selection,       \
+                                         params);                       \
+        int ret;                                                        \
+                                                                        \
+        if (key == NULL)                                                \
+            return 0;                                                   \
+                                                                        \
+        ret = impl##_##kind##2##output##_encode_o(ctx, key, cout,       \
+                                                  cb, cbarg);           \
+        ossl_prov_free_key(impl##_keymgmt_functions, key);              \
+        return ret;                                                     \
+    }                                                                   \
+    static int                                                          \
+    impl##_##kind##2##output##_encode_o(void *vctx, const void *key,    \
+                                        OSSL_CORE_BIO *cout,            \
+                                        OSSL_PASSPHRASE_CALLBACK *cb,   \
+                                        void *cbarg)                    \
+    {                                                                   \
+        int selection = type##_##kind##_selection;                      \
+                                                                        \
+        if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)         \
+            return key2any_encode(vctx, key, impl##_type_to_evp(key),   \
+                                  cout, key_to_##output##_pkcs8_bio,    \
+                                  cb, cbarg,                            \
+                                  prepare_##type##_params,              \
+                                  type##_priv_to_der);                  \
+        if ((selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) != 0)          \
+            return key2any_encode(vctx, key, impl##_type_to_evp(key),   \
+                                  cout, key_to_##output##_pubkey_bio,   \
+                                  cb, cbarg,                            \
+                                  prepare_##type##_params,              \
+                                  type##_pub_to_der);                   \
+        return key2any_encode_params(vctx, key, cout,                   \
```

Comment:
Dont know if this should be checking selection and return fail if not selected..

Reply:
You mean, if `selection` i zero?

Reply:
if selection was not SELECT_ALL_PARAMETERS != 0

Reply:
I added an assertion at the beginning.  Point is, if there are no params to be had, this line will never be reached, because the "params" variant will never be implemented.  That's ensure by `MAKE_ENCODER_NOPARAM`.
(I have more changes coming up, which affect this as well, so let's not complicate things too much, yeah?)